### DISCUSSION TITLE
LWIP AutoIP?

### DISCUSSION THREAD

**Discussion by MartinChisnall** [earlephilhower/arduino-pico#2289] [Q&A]
Is Auto IP functional within this Arduino Core port to RP2040 please?
That is, if no DHCP offer is received will it self assign itself an ip address?
(Wired ethernet to Wiznet W5500 chip)
I know it exists withinn LWIP but I can't see it included in the LWIPOPTS.h where I would expect to find it.
Thank you

**Comment by earlephilhower** [comment]
No, it's not included.  Unfortunately it doesn't really work with the Arduino `connected()` definition.  You could try editing the `lwipopts.h` in `tools/libpico` and rebuilding the SDK library.  Then you can probably hack in a call to the autoip_xxx calls wherever needed...
